src/components/TodoItem.tsx

```diff
@@ -0,0 +1,115 @@
+import { FC, FormEvent, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todo: Todo;
+  onDelete?: () => void;
+  onToggle?: () => void;
+  onRename?: (todo: Todo) => void;
+  isLoading: boolean;
+}
+
+export const TodoItem: FC<Props> = ({
+  todo,
+  onDelete = () => {},
```

Comment:
you can do it optional and check before calling it

but also, you can improve the current approach: create some `noOperation` function (you can create better name)) above component and use it as default value for each one